benchmarks/benchmark-polynomial-matrix-mul-fft.C

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (C) 2020 Cyril Bouvier, Pascal Giorgi
+ *
+ * Written by Cyril Bouvier <cyril.bouvier@lirmm.fr>
+ *            Pascal Giorgi <pascal.giorgi@lirmm.fr>
+ *
+ * ========LICENCE========
+ * This file is part of the library LinBox.
+ *
+ * LinBox is free software: you can redistribute it and/or modify
+ * it under the terms of the  GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * ========LICENCE========
+ */
+
+
+
+
+#include <iostream>
+#include <iomanip>
+using namespace std;
+
+//#define FFT_PROFILER
+#include <linbox/ring/modular.h>
+#include <linbox/randiter/random-prime.h>
+#include <linbox/randiter/random-fftprime.h>
+#include <givaro/zring.h>
+#include <recint/rint.h>
+#include <linbox/matrix/matrix-domain.h>
+#include <linbox/util/commentator.h>
+#include <linbox/util/timer.h>
+#include <linbox/matrix/polynomial-matrix.h>
+#include <linbox/algorithms/polynomial-matrix/polynomial-matrix-domain.h>
+#include <linbox/algorithms/polynomial-matrix/matpoly-mult-naive.h>
+#include <linbox/algorithms/polynomial-matrix/matpoly-mult-kara.h>
+
+#include <givaro/modular.h>
+#include <givaro/modular-extended.h>
+#include <fflas-ffpack/utils/args-parser.h>
+
+using namespace std;
+using namespace LinBox;
+using Givaro::Modular;
+using Givaro::ModularExtended;
+
+/* For pretty printing type */
+template<typename ...> const char *TypeName();
+template<template <typename ...> class> const char *TypeName();
+
+#define REGISTER_TYPE_NAME(type) \
+    template<> const char *TypeName<type>(){return #type;}
+
+REGISTER_TYPE_NAME(float);
+REGISTER_TYPE_NAME(double);
+REGISTER_TYPE_NAME(uint16_t);
+REGISTER_TYPE_NAME(uint32_t);
+REGISTER_TYPE_NAME(uint64_t);
+REGISTER_TYPE_NAME(uint128_t);
+REGISTER_TYPE_NAME(Modular);
+REGISTER_TYPE_NAME(ModularExtended);
+
+
+
+using namespace LinBox;
+
+template<typename Rand, typename Matrix>
```

Comment:
This should be just a call to `A.random(r)`. The method exists for `DenseMatrix`. If it does not exist for polynomail matrices is should be added.

Reply:
Right. I've just added the methods to the polynomial matrix classes and correct the program.